hnn_core/gui/gui.py

```diff
@@ -1229,66 +1309,82 @@ def _init_network_from_widgets(params, dt, tstop, single_simulation_data,
-                weights_nmda=weights_nmda,
-                synaptic_delays=synaptic_delays,
-                event_seed=drive['seedcore'].value)
+            synaptic_delays = {k: v.value for k, v in drive['delays'].items()}
+            print(
+                f"drive type is {drive['type']}, location={drive['location']}")
+            if drive['type'] == 'Poisson':
+                rate_constant = {
+                    k: v.value
+                    for k, v in drive['rate_constant'].items() if v.value > 0
+                }
+                weights_ampa = {
+                    k: v
+                    for k, v in weights_ampa.items() if k in rate_constant
+                }
+                weights_nmda = {
+                    k: v
+                    for k, v in weights_nmda.items() if k in rate_constant
+                }
+                single_simulation_data['net'].add_poisson_drive(
+                    name=drive['name'],
+                    tstart=drive['tstart'].value,
+                    tstop=drive['tstop'].value,
+                    rate_constant=rate_constant,
+                    location=drive['location'],
+                    weights_ampa=weights_ampa,
+                    weights_nmda=weights_nmda,
+                    synaptic_delays=synaptic_delays,
+                    space_constant=100.0,
+                    event_seed=drive['seedcore'].value)
+            elif drive['type'] in ('Evoked', 'Gaussian'):
+                single_simulation_data['net'].add_evoked_drive(
+                    name=drive['name'],
+                    mu=drive['mu'].value,
+                    sigma=drive['sigma'].value,
+                    numspikes=drive['numspikes'].value,
+                    location=drive['location'],
+                    weights_ampa=weights_ampa,
+                    weights_nmda=weights_nmda,
+                    synaptic_delays=synaptic_delays,
+                    space_constant=3.0,
+                    event_seed=drive['seedcore'].value)
+            elif drive['type'] in ('Rhythmic', 'Bursty'):
+                single_simulation_data['net'].add_bursty_drive(
+                    name=drive['name'],
+                    tstart=drive['tstart'].value,
+                    tstart_std=drive['tstart_std'].value,
+                    burst_rate=drive['burst_rate'].value,
+                    burst_std=drive['burst_std'].value,
+                    location=drive['location'],
+                    tstop=drive['tstop'].value,
+                    weights_ampa=weights_ampa,
+                    weights_nmda=weights_nmda,
+                    synaptic_delays=synaptic_delays,
+                    event_seed=drive['seedcore'].value)
+
+        elif drive['type'] == 'Tonic':
+            cell_types = list(drive['amplitude'].keys())
+            assert set(cell_types) == set(drive['t0'].keys()) == set(
+                drive['tstop'].keys())
+            for cell_type in cell_types:
```

Comment:
as we discussed above, I think we don't need this external loop ... it will greatly simplify your code and the interface. Just a dropdown for selecting the cell_type is needed

Reply:
I kind of forgot the details. could you correct me if I am wrong? are you saying that the bias has the same value for all cell types? And thus we don't need to let the user to set different values for different cell types

Reply:
I think we decided that the start and stop times should be the same for all cell types targetted by the tonic drive, but that this needs to be changed at the API-level within `Network.add_tonic_bias()`.

Reply:
For now I think we can simply link those components so that they look all the same. This is the most straightforward solution and visually reinforces to the user they are meant to be the same, rather than implying an error on our end.

Reply:
Check the latest commit, we just linked them alltogether

Reply:
Why not just fix `add_tonic_bias()` so that we don't have to revert all of this later?

Reply:
It should probably be it's own PR but shouldn't take too long. If I get a chance this week, I'm happy to open a PR.

Reply:
I agree with Ryan that we should add an option to specify the tonic biases using a dictionary so that the logic of network instantiating is not in the GUI. The GUI should be a thin layer on top of the computational core

Reply:
Unfortunately I never got around to this.

Yes, the idea was to be able to set which cell types are targeted in `Network.add_tonic_bias()` (similar to the way we set which cells are targeted by setting weights in `Network.add_evoked_drive()`). The best way to do with would probably be to remove the `cell_type` arg and modify the `amplitude` arg to be dict that maps amplitude to a given cell type key.

Reply:
we should certainly fix the API before adding to the GUI